onnxruntime/core/optimizer/unsqueeze_elimination.cc

```diff
@@ -31,8 +21,24 @@ Status UnsqueezeElimination::Apply(Graph& graph, Node& node, RewriteRuleEffect&
     return Status::OK();
   }
 
-  std::vector<int64_t> new_dims(axes.size() + tensor_proto.dims().size(), 0);
+  std::vector<int64_t> axes;
+  if (!graph_utils::GetRepeatedNodeAttributeValues(node, "axes", axes)) {
+    // missing 'axes'. should have failed at model load but just in case...
+    return Status::OK();
+  }
+
+  auto num_axes = axes.size();
+  auto output_rank = num_axes + tensor_proto.dims().size();
+
+  // handle any negative axis values
+  for (auto& axis : axes) {
+    if (axis < 0) {
+      axis += output_rank;
```

Comment:
Add some check whether axis is in proper range: For example, axis range could be [-r+1, r] where r = rank. If user input an axis -2r, then following code like "new_dims[axis]=1" could has problem.

Reply:
The axis range is validated in the ONNX shape inferencing so would fail at model load if this happened. Note that we're only doing the unsqueeze elimination on a constant initializer, so the shape is available during ONNX shape inferencing in order to do the validation at that point.

There's a test case that loads a model with a negative axis as part of this PR. Has values of -1 and -2 in the model.